x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React, { useState } from 'react';
+import {
+  EuiComboBox,
+  EuiFieldSearch,
+  // @ts-ignore
+  EuiHighlight,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiButtonEmpty,
+  EuiFilterGroup,
+  EuiFilterButton,
+  EuiContextMenuPanel,
+  EuiContextMenuItem,
+  EuiContextMenuPanelProps,
+  EuiPopover,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { DatasourceDataPanelProps, DataType } from '../types';
+import { IndexPatternPrivateState, IndexPatternField } from './indexpattern';
+import { ChildDragDropProvider } from '../drag_drop';
+import { FieldItem } from './field_item';
+import { FieldIcon } from './field_icon';
+
+// TODO the typings for EuiContextMenuPanel are incorrect - watchedItemProps is missing. This can be removed when the types are adjusted
+const FixedEuiContextMenuPanel = (EuiContextMenuPanel as any) as React.FunctionComponent<
+  EuiContextMenuPanelProps & { watchedItemProps: string[] }
+>;
+
+function sortFields(fieldA: IndexPatternField, fieldB: IndexPatternField) {
+  return fieldA.name.toLowerCase() < fieldB.name.toLowerCase() ? -1 : 1;
```

Comment:
I've been preferring `String.localeCompare` for this because lowercase is not the same in all languages: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare